checkov/terraform/graph_builder/foreach/abstract_handler.py

```diff
@@ -68,14 +68,21 @@ def _build_sub_graph(self, blocks_to_render: list[int]) -> TerraformLocalGraph:
         return sub_graph
 
     @staticmethod
-    def _update_nested_tf_module_foreach_idx(original_foreach_or_count_key: int | str, original_module_key: TFModule,
-                                             tf_moudle: TFModule | None) -> None:
-        original_module_key.foreach_idx = None  # Make sure it is always None even if we didn't override it previously
-        while tf_moudle is not None:
-            if tf_moudle == original_module_key:
-                tf_moudle.foreach_idx = original_foreach_or_count_key
-                break
-            tf_moudle = tf_moudle.nested_tf_module
+    def _get_module_with_only_relevant_foreach_idx(original_foreach_or_count_key: int | str,
```

Comment:
why not move this method to the `ForeachModuleHandler` it seems to be only relevant for it

Reply:
moved :)